### ISSUE TITLE
Love the library, small crash report

### ISSUE THREAD

**Issue by acrosa** [asg017/sqlite-vec#11]
I was able to successfully compile and integrate a static version of _sqlite-vec_ into my personal app (https://www.getrebound.xyz). However, I encountered an unexpected crash that I was able to reproduce using the `dylib` from the `latest v0.0.1-alpha.10` release and the sqlite client.

Selecting or trying to select or delete a row that doesn't exist crashes with an assertion.

```
sqlite> .load /path/vec0.dylib
sqlite> SELECT sql FROM sqlite_master WHERE type='table' AND name='vec_examples';
CREATE VIRTUAL TABLE vec_examples USING vec0(
    sample_embedding float[8]
)
sqlite> SELECT rowid, distance FROM vec_examples;
1|
2|
3|
...
93|
94|
95|
96|
97|
98|
100|
sqlite> SELECT rowid FROM vec_examples where rowid = 100;
100
sqlite> SELECT rowid, distance FROM vec_examples where rowid = 99;
Assertion failed: (rc == 100), function vec0_get_vector_data, file sqlite-vec.c, line 2685.
[1]    21973 abort      sqlite3 
```

Maybe it's a known issue. Very excited about this little library!

**Comment by franciscojavierarceo** [comment]
I also get a crash when I try to add another field to the virtual table. 

For example:

```sql
CREATE VIRTUAL TABLE vec_examples USING vec0(
    sample_embedding float[8]
    primary_key blob
)
```
results in a break.

**Comment by asg017** [comment]
Hey @acrosa , sorry for the delay - thanks for the report! This should be fixed now as of at least `v0.0.1-alpha.19`, but lmk if you still see it!
